### PR TITLE
[DSS-434]: feat() - Add `component-id` to components

### PR DESCRIPTION
**Pull request by pixelflips**
# Description

Adds `component-id` prop to all components that currently do not have it.
Updates `component-id` description copy so they match across all components.

- Avatar
- Button
- Chip
- Divider
- Image
- Link

Fixes [DSS-434](https://kajabi.atlassian.net/browse/DSS-434)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

1. Navigate to the components above and verify that `component-id` has been added and works as expected.
2. Run tests locally and verify that all tests pass.

- [ ] unit tests
- [x] e2e tests
- [ ] accessibility tests
- [x] tested manually
- [ ] other:

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

[DSS-434]: https://kajabi.atlassian.net/browse/DSS-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/core/src/components.d.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ export namespace Components {
           * @defaultValue false
          */
         "badge"?: boolean;
+        /**
+          * String used for the component `id` attribute.
```
Comment (ju-Skinner) [inline libs/core/src/components.d.ts]: The word "string" used here is redundant as the type is set to be "string".  I personally like the previous description that was used for `input`.

How is this, it's a meet-in-the-middle kind of approach. 
> A unique identifier used for the underlying component `id` attribute.

**Review by anechol — APPROVED** [review]
Thanks for adding all these for consistency.

File: libs/core/src/components.d.ts (hunk 1 above)
Comment (pixelflips) [reply libs/core/src/components.d.ts]: Works for me. I'll get them all updated.

Comment (pixelflips) [reply libs/core/src/components.d.ts]: Done.

**Review by ju-Skinner — APPROVED** [review]
Thank you, awesome job

**Pull request merged**